### PR TITLE
Refactor metrics transformation & make endpoint non-blocking

### LINKED ISSUE #135: Run queries against Prometheus concurrently
This could help to improve the speed of the metrics provider

### LINKED ISSUE #136: Make metrics endpoint async to eliminate delay on other endpoints

### PR DESCRIPTION
**Pull request by disrupted**
Closes #136 
Closes #135

### CODE REVIEW

File: backend/streams_explorer/core/services/metric_providers.py [hunk 1]
```diff
@@ -1,51 +1,65 @@
...
         "sum by(topic) (rate(kafka_topic_partition_current_offset[5m]))",
+        lambda result: {
+            d["metric"]["topic"]: round(float(d["value"][-1]), 2) for d in result
+        },
     )
     MESSAGES_OUT = (
         "messages_out",
         "sum by(topic) (rate(kafka_consumergroup_group_offset[5m]) >= 0)",
+        lambda result: {
+            d["metric"]["topic"]: round(float(d["value"][-1]), 2) for d in result
```
Comment (philipp94831) [inline backend/streams_explorer/core/services/metric_providers.py:31]: this lambda seems to be repeated quite often. Try to reuse it by extracting methods

**Review by philipp94831 — CHANGES_REQUESTED** [review]

File: backend/streams_explorer/core/services/metric_providers.py [hunk 2]
```diff
@@ -87,82 +101,47 @@ def update(self):
...
+            return await self._query(metric.query)
+        except PrometheusException as e:
             logger.error(f"Error pulling {metric}: {e}")
         return []
 
-    def __prom_request(self, query: str) -> List:
-        return self._prom.custom_query(query)
+    async def _query(self, query: str) -> list:
+        r = await self._client.get(f"{self._api_base}/query", params={"query": query})
+        if r.status_code == 200:
```
Comment (philipp94831) [inline backend/streams_explorer/core/services/metric_providers.py:128]: Pretty sure there is a constant for HTTP_OK somewhere

File: backend/streams_explorer/core/services/metric_providers.py (hunk 1 above)
Comment (disrupted) [reply backend/streams_explorer/core/services/metric_providers.py]: I created a Transformer class now to replace it

Comment (disrupted) [reply backend/streams_explorer/core/services/metric_providers.py]: I am also going to change the key

**Review by VictorKuenstler — APPROVED** [review]

**Comment by disrupted** [comment]
on second thought the Transformer can actually be part of the enum

**Pull request merged**